singleton/controller.go

```diff
@@ -27,6 +31,39 @@ func AsReconciler(reconciler Reconciler) reconcile.Reconciler {
 	})
 }
 
+type ChannalObjectReconciler[T client.Object] interface {
```

Comment:
This is so bespoke and specific to events, I'd honestly just apply this to the event controller alone and call it good. This is basically shimming a watch informer on top of controller-runtime. You could do this same thing with just creating a watch informer and firing a metric when you get an Create event on an event, but we're basically just wrapping this in controller-runtime's interface so that it adheres to the same interface for all controllers.

Unless n > 2, I wouldn't generalize

Reply:
> This is basically shimming a watch informer on top of controller-runtime.

Yeah, this was the only purpose for the reconciler.  I thought this might be useful as a reconciler in the future, but can drop it for now.